apps/entry/filter_set.py

```diff
@@ -643,6 +645,11 @@ def lead_group_label_filter(self, queryset, name, value):
             return queryset.filter(entrygrouplabel__group__title__icontains=value)
         return queryset
 
+    def lead_has_assessment_filter(self, qs, _, value):
+        if value is None:
+            return qs
+        return qs.filter(lead__assessmentregistry__isnull=not value)
```

Comment:
@thenav56 why not `return qs.filter(lead__assessmentregistry__isnull = value)`

Reply:
lead_has_assessment=True
-> lead__assessmentregistry needs to have value so lead__assessmentregistry__isnull=False

lead_has_assessment=False
-> lead__assessmentregistry needs to be empty so lead__assessmentregistry__isnull=True